plugins/amoeba/platforms/cuda/src/AmoebaCudaKernels.cpp

```diff
@@ -1791,6 +1812,48 @@ void CudaCalcAmoebaMultipoleForceKernel::getInducedDipoles(ContextImpl& context,
     }
 }
 
+
+void CudaCalcAmoebaMultipoleForceKernel::getTotalDipoles(ContextImpl& context, vector<Vec3>& dipoles) {
+    ensureMultipolesValid(context);
+    int numParticles = cu.getNumAtoms();
+    dipoles.resize(numParticles);
+    const vector<int>& order = cu.getAtomIndex();
+    if (cu.getUseDoublePrecision()) {
+        vector<double4> posqVec;
+        vector<double> labDipoleVec;
+        vector<double> inducedDipoleVec;
+        vector<double> totalDipoleVecX(numParticles);
+        vector<double> totalDipoleVecY(numParticles);
+        vector<double> totalDipoleVecZ(numParticles);
```

Comment:
I don't think you actually need the totalDipoleVec arrays, since you just set one element, use it immediately, and then never look at it again.  Those could just be local variables inside the loop.